src/game/Spell.cpp

```diff
@@ -5100,9 +5100,7 @@ SpellCastResult Spell::CheckCast(bool strict)
                     else
                     {
                         Pet* dbPet = new Pet;
-                        if (dbPet->LoadPetFromDB((Player*)m_caster, 0))
-                            return SPELL_CAST_OK;           // still returns an error to the player, so this error must come from somewhere else...
-                        else
+                        if (!dbPet->TryLoadFromDB((Player*)m_caster, 0))
```

Comment:
This must not be done here, because Effect sends SPELLOG, which will be implemented very soon.